### PR TITLE
added fix or pow(base, 0)

### PR DESCRIPTION
**Pull request by wanderer**

### CODE REVIEW

**Comment by wanderer** [comment]
this reverts and squashes the commits for crunch benchmark in #37 and adds a fix for pow(base, 0)

File: lib/bn.js [hunk 1]
```diff
@@ -2156,6 +2156,10 @@ Red.prototype.invm = function invm(a) {
 Red.prototype.pow = function pow(a, num) {
   var w = [];
   var q = num.clone();
+
+  if (num.cmpn(0) === 0)
```
Comment (dcousens) [inline lib/bn.js]: `num === q`, maybe do it before the clone?

Comment (wanderer) [inline lib/bn.js]: sure

**Comment by indutny** [comment]
LGTM, thanks!

**Comment by indutny** [comment]
Landed, thank you!

**Pull request closed without merging**